### PR TITLE
Change Proxy Header Regression tests into Catch unit tests.

### PR DESCRIPTION
**Pull request by ywkaras**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -123,6 +123,7 @@ proxy/http3/test_libhttp3
 proxy/http3/test_qpack
 proxy/logging/test_LogUtils
 proxy/logging/test_LogUtils2
+proxy/logging/test_RolledLogDeleter
```
Comment (ywkaras) [inline .gitignore]: A one-line change not actually related to the purpose of this PR.

File: include/tscore/Regression.h [hunk 2]
```diff
@@ -54,8 +54,6 @@
 #define REGRESSION_TEST_QUICK 1
 #define REGRESSION_TEST_NIGHTLY 2
 #define REGRESSION_TEST_EXTENDED 3
-// use only for testing TS error handling!
-#define REGRESSION_TEST_FATAL 4
```
Comment (ywkaras) [inline include/tscore/Regression.h]: A small change not actually related to the purpose of this PR.

**Comment by zwoop** [comment]
This will have to go into 9.0.x, or other PRs will have serious merge conflicts.

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**